### ISSUE TITLE
Bluetooth Eddystone sample does not correctly implement Eddystone beacon

### ISSUE THREAD

**Issue by nashif** [zephyriot/zep-jira14#566]
**_Reported by Henry Bruce:_**

*Expected behavior*
Run sample (on Arduino 101). Start beacon finder app  on smart phone that will detect Arduino 101 beacon and take you to https://www.zephyrproject.org/

*Observed behavior*
Run sample (on Arduino 101). Start beacon finder app  on smart phone but Arduino 101 bean not detected. GATT debug tool shows that Eddystone service UUID is incorrect

*Workaround*
Use the beacon sample instead, it correctly implements an Eddystone beacon. Suggest removing Eddystone sample.

(Imported from Jira ZEP-641)

**Comment by nashif** [comment]
**_by Luiz Augusto von Dentz:_**

And where are the logs? Btw we are following these spec:

https://github.com/google/eddystone/tree/master/configuration-service

Not this (which seems much older and with uses different UUIDs):

https://github.com/google/eddystone/blob/master/eddystone-url/docs/config-service-spec.md

**Comment by nashif** [comment]
**_by Henry Bruce:_**

* [Screenshot_2016-08-10-13-39-17.png](https://s3.amazonaws.com/logs.zephyrproject.org/jira/GH-566/Screenshot_2016-08-10-13-39-17.png) (size: 149847)

**Comment by nashif** [comment]
**_by Henry Bruce:_**

* [eddystone.log](https://s3.amazonaws.com/logs.zephyrproject.org/jira/GH-566/eddystone.log) (size: 2767)

**Comment by nashif** [comment]
**_by Henry Bruce:_**

* [gatttool.log](https://s3.amazonaws.com/logs.zephyrproject.org/jira/GH-566/gatttool.log) (size: 2031)

**Comment by nashif** [comment]
**_by Henry Bruce:_**

I'm not a GATT expert but it seems that the primary service UUID should be 0000FEAA-0000-1000-8000-00805F9B34FB as per https://github.com/google/eddystone/blob/master/protocol-specification.md. See also validator test tool at https://github.com/google/eddystone/blob/master/tools/eddystone-validator/EddystoneValidator/app/src/main/java/com/google/sample/eddystonevalidator/MainActivityFragment.java, line 75.

The eddystone sample only advertises the configuration service  (UUID a3c87500-8ed3-4bdf-8a39-a01bebede295) so it not being detected as an Eddystone beacon. The attached gatttool output, sample log and Android GATT sniffer app screen show all confirm this. 

The beacon sample advertises a UUID of 0000FEAA-0000-1000-8000-00805F9B34FB so is detected by Eddystone beacon scanners (including Google's Android Eddystone validator app).

Out of interest which beacon scanning tool did you use to validate? Maybe I'm misunderstanding the purpose of this sample.

**Comment by nashif** [comment]
**_by Luiz Augusto von Dentz:_**

0000FEAA-0000-1000-8000-00805F9B34FB is not a GATT service, the https://github.com/google/eddystone/tree/master/configuration-service is pretty clear about what it shall be advertise  when on configuration mode:

'Once your beacon becomes connectable, we recommend broadcasting the configuration GATT service's UUID as a 128-bit service UUID in the scan response, along with a local name data type that identifies your beacon.'

So while at this mode you should be able to configure it, the spec actually suggests to have a timeout of 30 seconds and if there is no connection in the meantime it will turn back into beacon mode which is something the sample is not doing but as far as configuration mode it seems correct, note that it will never turn into beacon mode while it is connected but once disconnected it shall advertise the configure URL.

**Comment by nashif** [comment]
**_by Luiz Augusto von Dentz:_**

I end up updating the sample with a timeout as suggested by eddystone specification:

https://gerrit.zephyrproject.org/r/#/c/4092/

Note: This is currently on bluetooth branch but will be merged to master once the merge window opens.

**Comment by nashif** [comment]
**_by Henry Bruce:_**

What beacon scanner / config software did you use to test your update?
I will validate once this is merged into master.

**Comment by nashif** [comment]
**_by Anas Nashif:_**

status?

**Comment by nashif** [comment]
**_by Mark Linkmeyer:_**

Fixing incorrect priority

**Comment by nashif** [comment]
**_by Dawei Wu:_**

_Henry Bruce_  , the latest zephyr code has timeout 30s for configuration and beacon mode switch , see below. So you can have a try , if it's OK for you we can close this . Thanks.
```
[bt] [WRN] set_static_addr: Using temporary static random address
[bt] [INF] show_dev_info: Identity: f8:12:a4:14:1f:dc (random)
[bt] [INF] show_dev_info: HCI: version 4.2 (0x08) revision 0x0000, manufacturer 0xffff
[bt] [INF] show_dev_info: LMP: version 4.2 (0x08) subver 0xffff
Bluetooth initialized
Configuration mode: waiting connections...
Switching to Beacon mode.

```

**Comment by nashif** [comment]
**_by Henry Bruce:_**

OK to close.

**Comment by nashif** [comment]
**_by Dawei Wu:_**

So close this.
